_rules/widget-non-empty-accessible-name-rdzs6q.md

```diff
@@ -0,0 +1,382 @@
+- [H44: Using label elements to associate text labels with form controls](https://www.w3.org/WAI/WCAG21/Techniques/html/H44)
+- [H65: Using the title attribute to identify form controls when the label element cannot be used](https://www.w3.org/WAI/WCAG21/Techniques/html/H65)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA16)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `input` element has an [accessible name][] because of its [programmatic label](#programmatic-label).
+
+```html
+<label>
+	first name
+	<input />
+</label>
+```
+
+#### Passed Example 2
+
+This `input` element has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<div>last name</div>
+<input aria-label="last name" disabled />
+```
+
+#### Passed Example 3
+
+This `select` element has an [accessible name][] because of its [programmatic label](#programmatic-label).
+
+```html
+<label for="country">Country</label>
+<select id="country">
+	<option>England</option>
+	<option>Scotland</option>
+	<option>Wales</option>
+	<option>Northern Ireland</option>
+</select>
+```
+
+#### Passed Example 4
+
+This `textarea` element has an [accessible name][] because of its `aria-labelledby` attribute.
+
+```html
+<div id="country">Country</div>
+<textarea aria-labelledby="country"></textarea>
+```
+
+#### Passed Example 5
+
+This `input` element has an [accessible name][] because of its `placeholder` attribute.
+
+```html
+<input placeholder="Your search query" /> <button type="submit">search</button>
+```
+
+#### Passed Example 6
+
+This [semantic][semantic role] `combobox` element has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<div>Country</div>
+<div aria-label="country" role="combobox" aria-disabled="true">England</div>
+```
+
+#### Passed Example 7
+
+This [semantic][semantic role] `checkbox` element has the text content as its [accessible name][].
+
+```html
+<div role="checkbox">I agree to the terms and conditions.</div>
+```
+
+#### Passed Example 8
+
+This `button` element has an [accessible name][] because of its text content.
+
+```html
+<button>My button</button>
+```
+
+#### Passed Example 9
+
+This `input` element has an [accessible name][] because of its `value` attribute.
+
+```html
+<input type="submit" value="Submit" />
+```
+
+#### Passed Example 10
+
+This element with a `button` role has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<span role="button" aria-label="My button"></span>
+```
+
+#### Passed Example 11
+
+This `input` element has an [accessible name][] because of the default accessible name for an `input` element with a `type` attribute set to `reset`.
```

Comment:
```suggestion
This `input` element has an [accessible name][] because of the default accessible name for an `input` element with a `type` [attribute value][] of `reset`.
```